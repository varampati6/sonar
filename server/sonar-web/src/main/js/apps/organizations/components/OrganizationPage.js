/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
// @flow
import React from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import OrganizationNavigation from '../navigation/OrganizationNavigation';
import { fetchOrganization } from '../actions';
import { getOrganizationByKey } from '../../../store/rootReducer';
import type { Organization } from '../../../store/organizations/duck';
import NotFound from '../../../app/components/NotFound';

type OwnProps = {
  params: { organizationKey: string }
};

type Props = {
  children?: React.Element<*>,
  location: Object,
  organization: null | Organization,
  params: { organizationKey: string },
  fetchOrganization: string => Promise<*>
};

class OrganizationPage extends React.PureComponent {
  mounted: boolean;
  props: Props;
  state = { loading: true };

  componentDidMount() {
    this.mounted = true;
    this.updateOrganization(this.props.params.organizationKey);
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.params.organizationKey !== this.props.params.organizationKey) {
      this.updateOrganization(nextProps.params.organizationKey);
    }
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  updateOrganization = (organizationKey: string) => {
    if (this.mounted) {
      this.setState({ loading: true });
    }
    this.props.fetchOrganization(organizationKey).then(() => {
      if (this.mounted) {
        this.setState({ loading: false });
      }
    });
  };

  render() {
    const { organization } = this.props;

    if (!organization || organization.canAdmin == null) {
      if (this.state.loading) {
        return null;
      } else {
        return <NotFound />;
      }
    }

    return (
      <div>
        <Helmet defaultTitle={organization.name} titleTemplate={'%s - ' + organization.name} />
        <OrganizationNavigation organization={organization} location={this.props.location} />
        {this.props.children}
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps: OwnProps) => ({
  organization: getOrganizationByKey(state, ownProps.params.organizationKey)
});

const mapDispatchToProps = { fetchOrganization };

export default connect(mapStateToProps, mapDispatchToProps)(OrganizationPage);

export const UnconnectedOrganizationPage = OrganizationPage;
