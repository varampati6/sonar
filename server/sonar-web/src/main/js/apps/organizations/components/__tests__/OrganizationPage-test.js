/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import { UnconnectedOrganizationPage } from '../OrganizationPage';

it('smoke test', () => {
  const wrapper = shallow(
    <UnconnectedOrganizationPage params={{ organizationKey: 'foo' }}>
      <div>hello</div>
    </UnconnectedOrganizationPage>
  );
  expect(wrapper).toMatchSnapshot();

  const organization = { key: 'foo', name: 'Foo', isDefault: false, canAdmin: false };
  wrapper.setProps({ organization });
  expect(wrapper).toMatchSnapshot();
});

it('not found', () => {
  const wrapper = shallow(
    <UnconnectedOrganizationPage params={{ organizationKey: 'foo' }}>
      <div>hello</div>
    </UnconnectedOrganizationPage>
  );
  wrapper.setState({ loading: false });
  expect(wrapper).toMatchSnapshot();
});

it('should correctly update when the organization changes', () => {
  const fetchOrganization = jest.fn(() => Promise.resolve());
  const wrapper = shallow(
    <UnconnectedOrganizationPage
      params={{ organizationKey: 'foo' }}
      fetchOrganization={fetchOrganization}>
      <div>hello</div>
    </UnconnectedOrganizationPage>
  );
  wrapper.setProps({ params: { organizationKey: 'bar' } });
  expect(fetchOrganization.mock.calls).toMatchSnapshot();
});
