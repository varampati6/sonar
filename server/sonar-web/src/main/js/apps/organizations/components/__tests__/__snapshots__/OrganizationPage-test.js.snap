// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not found 1`] = `<NotFound />`;

exports[`should correctly update when the organization changes 1`] = `
Array [
  Array [
    "bar",
  ],
]
`;

exports[`smoke test 1`] = `null`;

exports[`smoke test 2`] = `
<div>
  <HelmetWrapper
    defaultTitle="Foo"
    titleTemplate="%s - Foo"
  />
  <OrganizationNavigation
    organization={
      Object {
        "canAdmin": false,
        "isDefault": false,
        "key": "foo",
        "name": "Foo",
      }
    }
  />
  <div>
    hello
  </div>
</div>
`;
